vit_unetr_ssl/transforms.py

```diff
@@ -0,0 +1,224 @@
+        [
+            # Load image data using the key "image"
+            LoadImaged(keys=keys, image_only=True),
+            # Ensure that the channel dimension is the first dimension of the image tensor.
+            EnsureChannelFirstd(keys=keys),
+            # Ensure that the image orientation is consistently RPI
+            Orientationd(keys=keys, axcodes="RPI"),
+            # Resample the images to a specified pixel spacing
+            # NOTE: spine interpolation with order=2 is spline, order=1 is linear
+            Spacingd(keys=keys, pixdim=(1.0, 1.0, 1.0), mode=(2, 1)),
+            # Normalize the intensity values of the image
+            NormalizeIntensityd(keys=["image"], nonzero=False, channel_wise=False),
+            # Remove background pixels to focus on regions of interest.
+            # CropForegroundd(keys=["image"], source_key="image"),
+            # Pad the image to a specified spatial size if its size is smaller than the specified dimensions
+            ResizeWithPadOrCropd(keys=keys, spatial_size=spatial_size),
+            AsDiscreted(keys='label', to_onehot=None, threshold=0.5),
+            # Randomly crop samples of a specified size around the label (spinal cord)
+            # Note that it seems that the transform randomly selects a foreground point from image, then use it as
+            # center crop. This means that it can find the closest voxel that is just outside the SC and use it as the
+            # center (hence it includes the SC)
+            # https://github.com/Project-MONAI/MONAI/issues/452#issuecomment-636065502
+            RandCropByPosNegLabeld(
+                keys=keys,
+                label_key="label",
+                spatial_size=roi_size,
+                pos=2,
+                neg=1,
+                num_samples=2,
+                image_key="image",
+                image_threshold=0,
+            ),
+            # Create copies of items in the dictionary under new keys, allowing for the same image to be manipulated
+            # differently in subsequent transformations
+            CopyItemsd(keys=["image"], times=2, names=["gt_image", "image_2"], allow_missing_keys=False),
+
+            # AUGMENTED VIEW 1
+            OneOf(
+                transforms=[
+                    # Randomly drop regions of the image
+                    # 'dropout_holes=True': the dropped regions will be set to zero, introducing regions of no
+                    # information within the image.
+                    RandCoarseDropoutd(
+                        keys=["image"],
+                        prob=1.0,
+                        holes=number_of_holes,
+                        spatial_size=roi_size[0] // 4,
+                        dropout_holes=True,
+                        # max_spatial_size=(roi_size[0]//4, roi_size[1]//4, roi_size[2]//4)
+                    ),
+                    # 'dropout_holes=False': the areas inside the holes will be filled with random noise
```

Comment:
Likewise, this should be clarified as well! `dropout_holes=False` is just the opposite -- the `holes` will be kept and the rest of the image is dropped out 

Remember, we saw a full images with only the SC in the middel and the rest of the image was noise? So, I guess that the result of 2 successive transforms: (1) `dropout_holes=False` which drops out the entire image (except the holes) and fills the values with 0 (because the default value for `fill_value` is `0`) and (2) `RandCoarseShuffle` shuffles the pixels throughout the image.

Reply:
Again, good catch! You are right. I'll clarify the comments in future commits.

Reply:
Fixed in https://github.com/ivadomed/model-seg-dcm/pull/10/commits/7f2734084cdb1e0cdaa4e4890992351e0873653d and https://github.com/ivadomed/model-seg-dcm/pull/10/commits/bdb2ac155cb53b64423d7b7cb8e416492629707e